src/System.Private.Xml/src/System/Xml/NameTable.cs

```diff
@@ -176,6 +175,25 @@ public override string Get(char[] key, int start, int len)
             return null;
         }
 
+        internal string GetOrAddEntry(string str, int hashCode)
+        {
+            for (Entry e = _entries[hashCode & _mask]; e != null; e = e.next)
+            {
+                if (e.hashCode == hashCode && e.str.Equals(str))
+                {
+                    return e.str;
+                }
+            }
+
+            return AddEntry(str, hashCode);
+        }
+
+        internal static int ComputeHash32(string key)
+        {
+            ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(key.AsSpan());
```

Comment:
do you know why this isn't just calling `key.GetHashCode()`?

That does
```
        public override int GetHashCode()
        {
            return Marvin.ComputeHash32(ref Unsafe.As<char, byte>(ref _firstChar), _stringLength * 2, Marvin.DefaultSeed);
        }
```
Producing the same result.

Reply:
To avoid virtual dispatch?

Reply:
I don't remember the origin of that but I suppose that it is because same hash needs to be generated a char[] and it needs to match if it represents the same sequence of chars, so instead of treating it differently they opted to encapsulate this fact under NameTable.

Reply:
Yes, we need `char[]` overload too - I don't thing string.GetHashCode has the same seed we use in here.